operators/src/source/gdal_source.rs

```diff
@@ -370,6 +370,84 @@ impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
     }
 }
 
+#[derive(PartialEq, Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct GdalMetadataFixedTimes {
+    pub time_steps: Vec<TimeInterval>,
+    pub minimum_step: TimeStep,
+    pub time_placeholders: HashMap<String, GdalSourceTimePlaceholder>,
+    pub params: GdalDatasetParameters,
+    pub result_descriptor: RasterResultDescriptor,
+}
+
+#[async_trait]
+impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+    for GdalMetadataFixedTimes
+{
+    async fn loading_info(&self, query: RasterQueryRectangle) -> Result<GdalLoadingInfo> {
+        let valid_duration = TimeInterval::new(
+            self.time_steps[0].start(),
+            self.time_steps[self.time_steps.len() - 1].start(),
+        )?;
+
+        let parts = match valid_duration.intersect(&query.time_interval) {
+            Some(time_interval) => {
+                let mut loading_info_parts: Vec<GdalLoadingInfoPart> = Vec::new();
+
+                for (index, time_position) in self.time_steps.iter().enumerate() {
+                    if time_interval.contains(time_position) {
+                        loading_info_parts.push(
+                            self.params
+                                .replace_time_placeholders(&self.time_placeholders, *time_position)
+                                .map(|loading_info_part_params| GdalLoadingInfoPart {
+                                    time: *time_position,
+                                    params: loading_info_part_params,
+                                })?,
+                        );
+
+                        if index < self.time_steps.len() - 1
```

Comment:
There can also be gaps before the first element in `time_steps` and after the last one. Basically the whole `query.time_interval` has to be covered by the sum off all the `parts`.